packages/block-editor/src/components/inserter/index.js

```diff
@@ -40,12 +67,21 @@ const defaultRenderToggle = ( {
 		<Button
 			icon={ plusCircle }
 			label={ label }
+			display={ 'inline-flex' }
+			alignItems={ 'center' }
+			color={ 'dark-gray-500' }
+			background={ 'none' }
+			border={ 'none' }
+			outline={ 'none' }
+			padding={ 'medium' }
+			margin={ 0 }
```

Comment:
I get that we want these props for a primitive but I don't feel very good having these leak to a higher-level abstraction (UI components) I'd rather have variations in the component itself.

Reply:
I think we can really use your and other reviewers’ guidance at this point about how to make this better. We also had concerns about overwriting styles for variants like `isPrimary`, `isSecondary`, however, we found in many places that Button style is overridden in CSS in addition to using variants, like [here](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/button-group/style.scss) or [here](https://github.com/Automattic/jetpack/blob/64db1ea2618a2a77757c0be4c5c15dfeba1e2f2d/extensions/blocks/tiled-gallery/editor.scss#L79). The idea behind this is, being able to meet the same need in an x-platform way. Otherwise, those kinds of customizations will not work on mobile. We could refactor these Buttons that have overridden styles to use PrimitiveButton instead but it will require a lot of extra work. I really like what @pinarol  suggested to put all styling params like `margin`, `padding` (not sure about additionalStyles) etc. to one and call it `experimentalStyles`, in that way we will expose only one prop in public API, WDYT about it?

Reply:
>  however, we found in many places that Button style is overridden in CSS in addition to using variants, like here or here. 

For me instead of doing that, I prefer if we should be looking at why we did these overrides in the first place and avoid them entirely with better abstractions/components/variations. These overrides quickly lead to bugs anyway.

Reply:
This was something we were questioning as well, “should we allow overriding the styles already defined in variants”? The ideal answer to this is actually “no”, this is also how syled-system variants page recommends: https://styled-system.com/variants given that you are sharing a similar opinion I think we can remove the props that are already defined by variants. However, we’ll still need to provide some props to be able to position the Button in a x-platform way like [here](https://github.com/Automattic/jetpack/blob/64db1ea2618a2a77757c0be4c5c15dfeba1e2f2d/extensions/blocks/slideshow/editor.scss#L14) or [here](https://github.com/Automattic/jetpack/blob/64db1ea2618a2a77757c0be4c5c15dfeba1e2f2d/extensions/blocks/slideshow/editor.scss#L14). Positioning/layout related props need to be available among all x-platform components not just Button, and those are: space, layout, flexbox, position  from https://styled-system.com/api Otherwise we won’t be able to provide an x-platform component, wdyt?

Reply:
I'm not certain I understand exacly why we need the positionning props absolutely. There are alternatives, Layout components (flex containers aware of their children,toolbars...). I'm not saying we won't ever need these but I believe it's better to start with no CSS props at all and consider use-case per use-case if there are better alternatives.

Reply:
>There are alternatives, Layout components (flex containers aware of their children,toolbars...).

Agree, but sometimes it's not enough and the 'margin', 'width' etc are needed. At this moment they are set in CSS like [here](https://github.com/Automattic/jetpack/blob/64db1ea2618a2a77757c0be4c5c15dfeba1e2f2d/extensions/blocks/publicize/editor.scss#L69) but we are not able to do the same on mobile. So we would need to wrap the button into some container (View) and add the margins to the container which is not performant and add one more nesting level.

> I'm not saying we won't ever need these but I believe it's better to start with no CSS props at all and consider use-case per use-case if there are better alternatives.

That's actually a good idea to do less in that PR. We could get rid of these props and then add them if needed during refactoring. However, I still feel like adding the `spacing` props will be one of the first things that will happen. There is a small part of the list of components that overrides these styles:
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/date-time/style.scss#L37
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/dropdown-menu/style.scss#L59
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/guide/style.scss#L7
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/guide/style.scss#L61
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/notice/style.scss#L40
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/notice/style.scss#L80
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/placeholder/style.scss#L93
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/placeholder/style.scss#L107
https://github.com/WordPress/gutenberg/blob/4dd6a8c09191f2aac358b4a166a61d82fbda10b7/packages/components/src/toolbar-button/style.scss#L2

I also checked for example the `rebassjs` lib that is UI component library and they allow to set i.e margin to button that is `primary` https://rebassjs.org/button/

Reply:
Actually, we referred to the already existing css customizations for Button component while adding the new style props. However I am now discovering that we actually don't quite approve most of those customizations either. What I am confused about is, isn't this a good reason to lean towards to a more encapsulated API design? 

If we hide the css inside the component and manage styles via props we’ll have a possibility to deprecate/remove those which we don't want to be overridden. 

Currently we are discussing what to “expose” in the API and I think it is a very valuable discussion to have. But doesn’t current state already expose every overriding possibility by providing the className publicly in the API? Also as far as I see we are already tied to a specific class name(components-button), just to provide overriding capabilities I assume? 

If we look at it this way this PR actually isn’t extending what is already exposed. On the contrary I think it can be a first step to be in control of what to expose in the future. 

Having said that, I think it’d be good to group style related props under one `experimental` prop at this stage, that would make us more free about trying different possibilities.

Reply:
About having margins/width/height in the API: I am thinking what can be the alternative approach? 

 1. Designing the Button in such a way that always fills the container? 

This requires putting a container per every Button just to be able to set margins/width/height, thus, doesn’t feel right. 

2. Determining margins/width/height inside the implementation, so nobody would be able to override? 

And this doesn’t look like a realistic goal considering almost every usage of Button sets a different margin. Plus, I believe it’s a fair need to change the margins/width/height of the component.